apps/src/aiTutor/interactionsApi.ts

```diff
@@ -13,14 +13,19 @@ export async function savePromptAndResponse(
   interactionData: AITutorInteraction
 ) {
   try {
-    await fetch('/ai_tutor_interactions', {
+    const response = await fetch('/ai_tutor_interactions', {
       method: 'POST',
       headers: {
         'Content-Type': 'application/json',
         'X-CSRF-Token': await getAuthenticityToken(),
       },
       body: JSON.stringify(interactionData),
     });
+    if (!response.ok) {
+      throw new Error('Network response was not ok');
```

Comment:
Is there a reason we throw an error here but if there's an error retrieving the feedbacks we log to the metrics reporter?

Reply:
Ah, good catch. So here are the patterns we currently have...

```
export async function savePromptAndResponse() {
  try {
    const response = await fetch('/ai_tutor_interactions', {...});
    if (!response.ok) {
      throw new Error('...');
    }
    const data = await response.json();
    return data;
  } catch (error) {
    MetricsReporter.logError({...});
  }
}
```

```
export const fetchAITutorInteractions = async () => {
  ...
  try {
    const response = await fetch(url, {...});
    if (!response.ok) {
      throw new Error('...');
    }
    const data = await response.json();
    return data;
  } catch (error) {
    MetricsReporter.logError({...});
    return null;
  }
};
```

```
export async function saveFeedback(
  aiTutorInteractionId: number,
  feedbackData: FeedbackData
) {
  try {
    await fetch(`/ai_tutor_interactions/${aiTutorInteractionId}/feedbacks`, {...});
  } catch (error) {
    MetricsReporter.logError({...});
  }
}
```

So actually, even though `savePromptAndResponse` is reacting to HTTP status codes in addition to errors thrown by `fetch` all of the functions here end execution within the catch block after logging the error. I'm trying to decide if that's okay -- I should probably file a follow-up task to make sure all this error handling is what we want.

Reply:
yeah totally fine to handle as a follow up